src/actionCreators/resourceHelpers.js

```diff
@@ -87,15 +87,20 @@ export const addResourceFromDataset =
   (dispatch) => {
     const subjectTerm = rdf.namedNode(chooseURI(dataset, uri))
     const newUri = asNewResource ? null : uri
-    const usedDataset = rdf.dataset()
-    usedDataset.addAll(
-      dataset.match(
+    const context = {
```

Comment:
I declared bankruptcy on passing around to many params and moved some to context.